### ISSUE TITLE
Unexpected Exception: maximum recursion depth exceeded in cmp

### ISSUE THREAD

**Issue by jvgutierrez** [anthcourtney/ansible-role-cis-amazon-linux#1]
Using ansible==2.2.0.0 and the following playbook against Amazon Linux 2016.09:
```

---

- hosts: localhost
  connection: local
  gather_facts: true
  become: yes

  vars:
    cis_level_1_exclusions:
      - 1.3.1
      - 1.3.2
      - "1.8"
      - 5.4.4
      - 3.4.2
      - 3.4.3
      - 6.2.13
    cis_pass_max_days: 45
    cis_umask_default: 002

  roles:
    - anthcourtney.cis-amazon-linux
```

I get the following error on task 2.1.3:
```
TASK [anthcourtney.cis-amazon-linux : include] *********************************
included: /home/ec2-user/anthcourtney.cis-amazon-linux/tasks/level-1/2.1.3.yml for localhost
ERROR! Unexpected Exception: maximum recursion depth exceeded in cmp```

If I add the task 2.1.3 to the exclusion list then I have the same issue with task 2.1.4.

I've attached the full traceback
[error.log.txt](https://github.com/anthcourtney/ansible-role-cis-amazon-linux/files/575106/error.log.txt)

**Comment by anthcourtney** [comment]
Many thanks for raising this. I'll look into it in the next couple of days.

On the face, it looks very similar to this bug from 2015 in ansible.

https://github.com/ansible/ansible/issues/12260

**Comment by jvgutierrez** [comment]
if it's useful, with ansible 2.0.2.0 works smoothly :)

**Comment by jdbass** [comment]
Same behaviour here with ansible 2.2.0.0 and Amazon Linux 2016.09, but only when calling with -t to try to limit sections.  I say try because it runs all sections even with -t section-1 or other section specified. 
Running the full playbook without -t it runs fine for me.

**Comment by anthcourtney** [comment]
Apologies for the delay, but I'm actively investigating this now.

**Comment by anthcourtney** [comment]
I can replicate this behaviour in ansible 2.2, and verify that it is not present in ansible 2.1.3

I've added some tests to this branch:

https://github.com/anthcourtney/ansible-role-cis-amazon-linux/tree/feature/ansible-2.2-support

which effectively apply the role to an Amazon Linux-esque docker container using both ansible 2.1.3 and ansible 2.2. The error only occurs in the ansible 2.2 container.

This appears to be a python-related issue (relating to sys.setrecursionlimit).

I think the only workaround possible is to reduce the number of includes, which would mean squashing individual checks into a single level-related file. (i.e. all 2.2.x files concatenated into a single 2.2.yml).

**Comment by anthcourtney** [comment]
I believe that I've fixed this in this change:

https://github.com/anthcourtney/ansible-role-cis-amazon-linux/commit/a982cffa81e85f0913639aaa4aafef963396938f

This also seems to have dramatically improved the speed of the role.

Would you mind testing this branch? If it works, I'll merge it to master and create a new tag, and then update the role on ansible galaxy.

**Comment by jdbass** [comment]
Runs a lot cleaner for me now thanks.  Avoiding the dynamic includes has dropped memory usage down hugely as well, down from ~700MB to 40MB when running against localhost.  No obvious issues from running a few quick tests.

**Comment by anthcourtney** [comment]
Released as v1.1.0 and pushed to ansible galaxy.
